Reference/google/protobuf/unittest_custom_options.pb.swift

```diff
@@ -1013,7 +1093,56 @@ public struct ProtobufUnittest_ComplexOptionType3: ProtobufGeneratedMessage {
     "complexoptiontype5": 2,
   ]}
 
-  var unknown = ProtobufUnknownStorage()
+  private class _StorageClass {
```

Comment:
Because one of its fields is a message type. The generator wasn't completely handling this correctly before.

Think of it this way: messages are value types, and `Optional<>` is a value type (an enum with non-`indirect` cases). Therefore, they cannot directly or indirectly have a circular reference back to the original struct. But it's possible to imagine a series of proto message definitions like this:

```protobuf
message A { B x = 1; }
message B { C f = 1; }
message C { D f = 1; }
...
message Y { Z f = 1; }
message Z { A f = 1; }
```

...which would fail if there was no heap storage in the middle to break the cycle, whether the generated properties were optional or not.